packages/toolkit/config/fast-refresh.php

```diff
@@ -18,6 +18,85 @@
 	);
 }
 
+if ( ! function_exists( __NAMESPACE__ . '\\get_block_asset_url' ) ) {
+	function get_block_asset_url( $path ) {
+		static $template_paths_norm = array();
+
+		$template = get_template();
+		if ( ! isset( $template_paths_norm[ $template ] ) ) {
+			$template_paths_norm[ $template ] = wp_normalize_path( get_template_directory() );
+		}
+
+		if ( str_starts_with( $path, trailingslashit( $template_paths_norm[ $template ] ) ) ) {
+			return get_theme_file_uri( str_replace( $template_paths_norm[ $template ], '', $path ) );
+		}
+	}
+
+	add_filter(
+		'block_type_metadata_settings',
+			static function( $settings, $metadata ) {
+				if ( isset( $metadata['editorStyle'] ) && str_starts_with( $metadata['editorStyle'], 'file:' ) ) {
+					$file              = $metadata['file'];
+					$dir               = dirname( $file );
+
+					if ( isset( $metadata['editorStyle'] ) ) {
+						$editor_style_path = $metadata['editorStyle'];
+						$editor_style_file_name = basename( $editor_style_path );
+
+						$js_editor_hmr_file = $dir . '/' . str_replace( '.css', '.js', $editor_style_file_name );
+
+						// TODO: do not enqueue this file if its name ends up being the same as editorScript
```

Comment:
I'm sure there might be some other edge cases, we need a way to safely load these things automatically, perhaps we add a flag per block?